src/Visualization/Python/PlotAlongLine.py

```diff
@@ -0,0 +1,249 @@
+#!/usr/bin/env python
+
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+import logging
+import os
+
+import click
+import matplotlib.pyplot as plt
+import numpy as np
+import rich
+
+import spectre.IO.H5 as spectre_h5
+from spectre.IO.Exporter import interpolate_to_points
+
+logger = logging.getLogger(__name__)
+
+
+def _parse_point(ctx, param, value):
+    if not value:
+        return None
+    return np.array(list(map(float, value.split(","))))
+
+
+@click.command(name="plot-along-line")
+@click.argument(
+    "h5_files",
+    nargs=-1,
+    type=click.Path(exists=True, file_okay=True, dir_okay=False, readable=True),
+)
+@click.option(
+    "--subfile-name",
+    "-d",
+    help="Name of subfile within h5 file containing volume data to plot.",
+)
+@click.option(
+    "--list-vars",
+    "-l",
+    is_flag=True,
+    help="Print available variables and exit.",
+)
+@click.option(
+    "--var",
+    "-y",
+    "vars",
+    multiple=True,
+    help=(
+        "Variables to plot. List any tensor components "
+        "in the volume data file, such as 'Shift_x'."
+    ),
+)
+@click.option(
+    "--step",
+    type=int,
+    help="Observation step number. Specify '-1' for the last step in the file.",
+)
+@click.option(
+    "--output",
+    "-o",
+    type=click.Path(file_okay=True, dir_okay=False, writable=True),
+    help=(
+        "Name of the output plot file. If unspecified, the plot is "
+        "shown interactively, which only works on machines with a "
+        "window server."
+    ),
+)
+@click.option(
+    "--line-start",
+    "-A",
+    callback=_parse_point,
+    help="Coordinates of the start of the line through the volume data.",
```

Comment:
Could you specify that these must be a comma separated list? Also, I think this differs from how we specify the origin in `Clip.py`. We should probably make these consistent.

Reply:
Added the comment. In `Clip.py` this is hard-coded to 3D so we can use `nargs=3`, but here the dimension is determined from the data files so `nargs` doesn't work and we can only check the number of coordinates later.

Reply:
What to you think about changing `Clip.py` (not necessarily in this PR) to use comma separated lists as well and then checking the number of args? I think it looks a bit cleaner on the command line